File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals,(a,b)->Integer.compare(a[0],b[0]));
        int n=intervals.length;
        ArrayList<int []> ar=new ArrayList<>();
        int first=intervals[0][0];
        int last=intervals[0][1];
        for(int a[]:intervals)
        {
            if(a[0]<=last)
            {
                last=Math.max(last,a[1]);
            }
            else
            {
                ar.add(new int[]{first,last});
                last=a[1];
                first=a[0];
            }
        }
        ar.add(new int[]{first,last});
        return ar.toArray(new int[ar.size()][]);        
    }
}

File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        int n=intervals.size();
        sort(intervals.begin(),intervals.end(),[](vector<int> a,vector<int> b){return a[0]<b[0];});
        vector<vector<int>> v;
        v.push_back(intervals[0]);
        for(int i=1;i<n;i++)
        {
            if(v.back()[1]>=intervals[i][0])
            {
                v.back()[1]=max(v.back()[1],intervals[i][1]);
            }
            else
            v.push_back(intervals[i]);
        }
        return v;
    }
};